### PR TITLE
allow manipulation of www compress behaviour

### PR DESCRIPTION
**Pull request by gordonb3**
Definitely something that belongs in the advanced users guide. Is
CPU more important, or is bandwidth? This switch allows the user to
make a different choice than what is considered to be sane for most
systems.

### CODE REVIEW

File: main/domoticz.cpp [hunk 1]
```diff
@@ -171,6 +172,7 @@ bool g_bStopApplication = false;
 bool g_bUseSyslog = false;
 bool g_bRunAsDaemon = false;
 bool g_bDontCacheWWW = false;
+signed char g_wwwCompressMode = 0x1; // 0x1 = on, 0x0 = static, 0xFF = off
```
Comment (gizmocuz) [inline main/domoticz.cpp]: could you make this an enum please

File: main/domoticz.cpp [hunk 2]
```diff
@@ -884,6 +886,21 @@ int main(int argc, char**argv)
...
+		{
+			_log.Log(LOG_ERROR, "Please specify a compress mode");
+			return 1;
+		}
+		std::string szmode = cmdLine.GetSafeArgument("-wwwcompress", 0, "on");
+		if (szmode == "off")
+			g_wwwCompressMode = 0xFF;
+		else if (szmode == "static")
+			g_wwwCompressMode = 0x0;
+		// should I verify possible invalid input? doesn't seem to be done with other parameters
```
Comment (gizmocuz) [inline main/domoticz.cpp]: Please do not add comments/questions in a PR
Always good to discuss it
No you do not as you set the default at the top of the code

File: main/domoticz.cpp (hunk 1 above)
Comment (gizmocuz) [reply main/domoticz.cpp]: Plz (also) make (it) a member of myWebem, and add a call to set the compression mode
that way you also do not have to work with externals in the webserver class which might be a stand-alone library later

**Review by gizmocuz — COMMENTED** [review]
Few small remarks

File: main/domoticz.cpp (hunk 2 above)
Comment (gordonb3) [reply main/domoticz.cpp]: So invalid equals default? As the comment states: parameter validation doesn't seem to be done on other inputs as well. Was more of a remark to myself and didn't really think about it when entering the PR (which in this case is mostly cut-and-paste from the original PR)

File: main/domoticz.cpp (hunk 1 above)
Comment (gordonb3) [reply main/domoticz.cpp]: less straightforward. need to rethink the decision points.

File: webserver/cWebem.h [hunk 3]
```diff
@@ -19,6 +19,12 @@ namespace http {
 			AUTH_LOGIN=0,
 			AUTH_BASIC,
 		};
+		enum _eWebCompressionMode
+		{
+			WWW_USE_GZIP=0,
```
Comment (gizmocuz) [inline webserver/cWebem.h:24]: Could you make no-compression value 0 ?

Comment (gizmocuz) [reply webserver/cWebem.h:24]: Is this enum public ? Then please use it everywhere (g_wwwCompressMode, in SetWebCompressionMode)

**Review by gizmocuz — COMMENTED** [review]
Could you consider the following changes

File: webserver/cWebem.h (hunk 3 above)
Comment (gordonb3) [reply webserver/cWebem.h:24]: It is public, but I need to transfer the value from domoticz.cpp to mainworker.cpp before I can use it. This is effectively the same method used for setting authentication method and webroot.

Comment (gordonb3) [reply webserver/cWebem.h:24]: There is actually no real logic to the enum. It merely seemed appropriate to use 0 for the default. If it is to be about (not) doing something special than the logical choice would be for `static` to be 0 as this is the setting that does not actively zip or unzip files.

Comment (gizmocuz) [reply webserver/cWebem.h:24]: Correct, the Authentication method should also be an enum

0 = Off, or Disabled , not On or Enabled

Comment (gordonb3) [reply webserver/cWebem.h:24]: I think it is? The two available choices are actually auth_login and auth_basic.

Comment (gizmocuz) [reply webserver/cWebem.h:24]: Now it is:
https://github.com/domoticz/domoticz/commit/5a00ba936ab9b39606add5ab7a783297fcd1446f

Comment (gordonb3) [reply webserver/cWebem.h:24]: Haha... I did not want to question existing constructs that were outside the scope of this PR. Hence it seemed appropriate to use the same method as there would likely be a good reason for it.

**Comment by gordonb3** [comment]
Looks like the build system is messed up?

File: webserver/cWebem.h (hunk 3 above)
Comment (gizmocuz) [reply webserver/cWebem.h:24]: I know, but this is how it was supposed to do, and could inspire you to also use enums !

File: main/domoticz.cpp [hunk 4]
```diff
@@ -171,6 +172,7 @@ bool g_bStopApplication = false;
 bool g_bUseSyslog = false;
 bool g_bRunAsDaemon = false;
 bool g_bDontCacheWWW = false;
+_eWebCompressionMode g_wwwCompressMode;
```
Comment (gizmocuz) [inline main/domoticz.cpp]: Could you set the default here (http::server::WWW_USE_GZIP)
Then the 'else' if (cmdLine.HasSwitch("-wwwcompress")) here is not needed

**Review by gizmocuz — COMMENTED** [review]
Could you add the default compression mode
This is also needed as when there is no -wwwcompress command line specified it is undefined.
(it's also good to always initialize variables)

File: main/domoticz.cpp (hunk 4 above)
Comment (gordonb3) [reply main/domoticz.cpp]: Aha! ***That*** is why I had it initialized to 0 when it was still an int...

Comment (gizmocuz) [reply main/domoticz.cpp]: Yes, that's why it is good everything is triple checked ;)

**Comment by gizmocuz** [comment]
Thanks!!

**Pull request merged**